includes help for admins and moderators

chrislennon opened the pull request:
This is a WIP PR relating to #189 .

@LuaMilkshake I was told you broke this originally so if you fix the conditions on line 48 and 52 which are conflicting the command should work.

chrislennon committed:
includes help for admins and moderators

jmariondev commented on commands/help.js:
```diff
@@ -28,16 +28,31 @@ module.exports = {
-        continue;
-      }
+      if (cmd.requireRoles){
+        let member = bot.guilds.first().members.get(message.author.id);
```
My thought here was to search through the bot's guilds and look for any of them that a user has special perms on to determine if it should include privileged commands in help output. I know *our* bot only belongs to one guild, but that's be more **the right way™**.

chrislennon replied:
In a recent branch I was handling it as a config/env var. @ThomasShaped advised against it at the time, but think mainly cause we had the above code everywhere else.

Should we move this logic to more of a `bot.guilds.find('name',appConfig.GUILD_NAME)` approach?
If so I can make the changes throughout on another PR.

jmariondev replied:
I suppose we can if we resign ourselves to being a one-guild bot (which we are, to be fair).

jmariondev commented on commands/help.js:
```diff
@@ -28,16 +28,31 @@ module.exports = {
+        console.log('command requires roles');
+        console.log('user has Admin: '+member.roles.exists('name','Admin')+ ' and is '+member.roles.exists('name','Admin'));
+        console.log('user has Moderator: '+member.roles.exists('name','Moderator')+ ' and is '+member.roles.exists('name','Moderator'));
+        if (cmd.requireRoles.includes('Admin') && !member.roles.exists('name','Admin')) {
```
This check would go below the check for Moderator, since commands have `requireRoles: ["Admin", "Moderator"]`, if a user is only a mod they'll get punted by this check.

chrislennon committed:
when in doubt, do a rewrite :sunglasses:

chrislennon committed:
suggested fixes

*embeds prefix command into codeblock (fixes 5+ msg lag)
*went markdown for some colour
*will not show moderator commands if admin

otter-computer commented on commands/help.js:
```diff
@@ -78,24 +83,22 @@ module.exports = {
     const userMessage = 'Available Commands';
```
I think this is not used anymore  😱

chrislennon committed:
unused variable

https://media.giphy.com/media/XqpnXaeZPnupy/giphy.gif
